Initial Test suite

stain88 opened the pull request:
Throwing up an initial set of tests.
Notes:
hitting the POST routes result in an actual message being sent, so these are marked as skip;
the following variables are used, and set up in `.env`:

- SLACK_USER_ID: your user ID
- SLACK_DM_ID: your DM ID, different to above
- SLACK_DM_NAME: your username
- SLACK_OTHER_USER_ID: another user ID
- SLACK_OTHER_DM_ID: other user's DM ID
- SLACK_OTHER_DM_NAME: other user's name
- SLACK_GENERAL_ID: ID for General
- SLACK_OTHER_CHANNEL_ID: any other channel ID

stain88 committed:
Initial basic test setup

stain88 committed:
Add user_id and slack_ver_token as test globals

stain88 committed:
setup for /welcome tests

stain88 committed:
check default /welcome with no parameters posts to user as DM

stain88 committed:
Remove trailing console.log

stain88 committed:
Export various channels/users

stain88 committed:
Remove parser test from welcome test

stain88 committed:
parser tests for /welcome

stain88 committed:
argument sorting tests for /resources

nalbina commented on test/helper.js:
```diff
@@ -0,0 +1,21 @@
+const other_channel_id = process.env.SLACK_OTHER_CHANNEL_ID;
+const slack_ver_token = process.env.SLACK_VERIFICATION_TOKEN;
+
+global.greetbot = greetbot;
```
Why do you need to make them global?

stain88 replied:
global is so that each test can pick them up, without the need to redefine/require in each file. Not sure if this was where I got it from, but similar issue: https://stackoverflow.com/questions/31282839/how-to-properly-require-modules-from-mocha-opts-file

nalbina commented on test/helper.js:
```diff
@@ -0,0 +1,21 @@
+const greetbot = require('../src/index.js');
+const user_id = process.env.SLACK_USER_ID;
```
maybe you could also do something like
```
const {
SLACK_USER_ID,
SLACK_DM_ID,
... } = process.env;
```
that way you dont need the variable declarations

nalbina commented on test/parser.test.js:
```diff
@@ -0,0 +1,176 @@
+    })
+
+    it('should target the user_id', async function() {
+      await parsePayload(req).then(res => {
```
```
const res = await parsePayload(req);
expect(res.target_channel_id[0]).to.equal(user_dm_id);
```

stain88 replied:
oh, yeah, that is neater

stain88 committed:
cleanup res in parser.test

stain88 committed:
simplify test variable declaration

angelocordon committed:
Add ESLint

Integrate ESLint to help us catch errors and write better JS.

angelocordon committed:
Config testing suite

Remove unused helper file; instead of using environment variables within a
testing suite, use a stubbed request instead.

angelocordon committed:
Rename incomingParser as parsePayload

Rename incomingParser due to how we semantically require it other modules.
Add top level documentation as to what parsePayload is for.

angelocordon committed:
Refactor parser test as parsePayload test

Renamed to match the same name as the module it's related to.
Stub request instead of using environment variables for simpler tests.
Reword some of the tests for easier readability when test is run.

angelocordon committed:
Clean up files based on ESLint recommendations

angelocordon committed:
Revert parsePayloadJS to incomingParser

Temporarily reverting until further discussion; moved back to its original
filename due to the area of coverage where it's already being used.

angelocordon committed:
Interpolate path instead of concatenating

Handles ESLint error: https://eslint.org/docs/rules/no-path-concat

angelocordon committed:
Handle error callbacks in routes test

angelocordon committed:
Skip welcome test

Comment out test code based on some discussions regarding shutting off
the `/welcome` command.

stain88 committed:
Merge pull request #66 from codebuddies/chore/update-test-suites

Update test suites [WIP]

stain88 committed:
rename action_arguments to actionArguments

stain88 committed:
fix skipped tests

stain88 committed:
cleaner fix for skipped tests

stain88 commented:
Fix #7 if this works

angelocordon committed:
Clean up incomingParser test

Remove unnecessary notes and comments.
Define `req`.
Lint syntax issues.
